### ISSUE TITLE
Script to automate payroll expenses

### ISSUE THREAD

**Issue by alanna** [opencollective/opencollective#6173]
## User story 
This is a request/suggestion from Scott, our accountant. At the moment we have a very manual and complicated process of creating expenses for payroll, where @alinamanko manually exports data from Justworks, sums up a bunch of different kinds of costs (wages, tax, benefits etc) and creates expenses to reflect how the funds need to move from the ops budget to payroll, and from Collectives to the ops budget to reimburse those costs in cases where it's a Collective employee. That's already a complication and time consuming process for Alina, but on top of that, to meet accounting requirements, Scott then has to identify and remove these entries from the data exports, because in Quickbooks we have to break it back down into the different costs (wages, taxes, etc) to be legally compliant for audits, etc. It's way too much to ask Alina or anyone to manually break all this down when creating the OC expenses. Even in the current method, it's error prone and overly complicated, so we can't add more steps.

## Best solution for this problem
A script which ingests the uploaded Justworks data export and automatically creates OC expenses with all the detailed breakdowns that we need to record (which are in the Justworks data), kind of similar to the one we created for Github Sponsors. I do not believe we should invest in a real integration between OC and Justworks, as different hosts use different payroll providers and it wouldn't be universally useful. We would ask Alina, Scott, and a developer to work together to clarify exactly how the script should work.

## MVP
A script is already MVP

## Metrics 
Time savings for Alina and Scott, fewer payroll related errors, reduced complexity

  P2 low frequency, high impact

**Comment by alanna** [comment]
@poohlaga would this be useful to OSC? Or do you need something similar for your other payroll provider's system?

**Comment by alinamanko** [comment]
Thank you @alanna - I agree that this would make life a whole lot easier now that the number of collectives uptaking the employment offering is on the rise. To give you an idea what I currently use, I am including an example extract for all the payments made during October. You will find there the raw data and a pivot summarizing the fields actually used. It will be a 'department' to identify the collective, and the 'amount'. The 'reference' field with the employee name is optional however it is used to determine the amount of admin fees OCF charges (which is a function of the number of employees managed). This may be the extra point of complexity which we don't look at for the GitHub export script and something we need to think about how to integrate @alanna. 

[Example export from Justworks - Oct.xls](https://github.com/opencollective/opencollective/files/10027684/Example.export.from.Justworks.-.Oct.xls)

**Comment by alinamanko** [comment]
OSC will have similar type of reports and currently the employment to collective is on hold but it will be really useful once OSC actively makes that offering.

**Comment by alinamanko** [comment]
@alanna one thing I would like to add here as well is that recent update about ParentPreneur. How do we ensure the script picks up a case where the cost are to be recorded on a project level rather than the general collective balance. I wonder if there is an easy way to flag it on Justwork perhaps? cc @natehn

**Comment by natehn** [comment]
_A request from a Collective admin:_

To make things easier in the future - especially knowing that we'll have more than one person to pay every month - could you please:

```
Specify the name of the person who's receiving the money (Mr. X, Ms. Y etc.)
Specify the period that is being paid
Submit the expenses to me so that I can approve it before it gets paid
```

That way, I think it should be a lot easier for us to keep track of each collaborators' payment.

**Comment by alanna** [comment]
Yes, except the part about them approving it. it's pre-approved by them signing up employees and we can't have a situation where it doesn't get approved and we're paying payroll and not debiting their budget.
